indexing-service/src/test/java/org/apache/druid/indexing/common/task/IndexTaskTest.java

```diff
@@ -2690,6 +2691,72 @@ public void testErrorWhenDropFlagTrueAndOverwriteFalse() throws Exception
     );
   }
 
+  // If shouldCleanup is false, cleanup should be a no-op
+  @Test
+  public void testCleanupIndexTask() throws Exception
+  {
+    new IndexTask(
```

Comment:
Could you mention in a comment that since taskLockBox is empty, we would throw a NPE if shouldCleanUp is true.